docs/developing.md

```diff
@@ -280,6 +284,55 @@ NOTE: The cluster name and domain used to create the certificate must match the
 Example:
 `hiveutil create-cluster mycluster --serving-cert=$HOME/mycluster.crt --serving-cert-key=$HOME/mycluster.key` 
 
+## Code editors and multi-module repositories
+
+Hive is a [multi-module repository](https://github.com/golang/go/wiki/Modules#faqs--multi-module-repositories) with
+the various submodules.
+
+`gopls` currently doesn't officially support multi-module repositories. There are current recommendations for some
+code editors to work around the current issue.
+
+See [gopls doc](https://github.com/golang/tools/blob/master/gopls/doc/workspace.md#multiple-modules) for next steps.
+
+## Updating Hive APIs
+
+Hive is a [multi-module repository](https://github.com/golang/go/wiki/Modules#faqs--multi-module-repositories) with
+the `github.com/openshift/hive/apis` go submodule.
+
+A separate `github.com/openshift/hive/apis` submodule allows other repositories to easily import the Hive APIs without
+having to deal with the complex vendoring introduced by dependencies of Hive controllers like the OpenShift Installer.
+
+Since Hive module `github.com/openshift/hive` or `root` module uses vendor directory for all it's dependencies, **ANY**
+change in `github.com/openshift/hive/apis` module requires updating the dependency in the `root` module for change to
+take effect.
+
+Therefore, these steps must be followed when updating the Hive APIs,
+
+1. Make necessary modifications to `github.com/openshift/hive/apis`.
```

Comment:
I tested the previous workflow with a make update, and was pleased to see the CRD yaml gets updated as before. would it be feasible to hook up an update of this specific dependency into make update such that it's done automatically if we follow this workflow normally?

Really nice to see that make verify also catches that I did not revendor.

Reply:
> would it be feasible to hook up an update of this specific dependency into make update such that it's done automatically if we follow this workflow normally?

@dgoodwin so when someone runs `make update`, we want the apis submodule updated in vendor ? Or some lite version of the vendor update?

Reply:
> I tested the previous workflow with a make update, and was pleased to see the CRD yaml gets updated as before. 

@dgoodwin i'm hoping this meant when you updated something in apis/ and ran `make update` the crds were updated like it did before the submodules?

Reply:
> > I tested the previous workflow with a make update, and was pleased to see the CRD yaml gets updated as before.
> 
> @dgoodwin i'm hoping this meant when you updated something in apis/ and ran `make update` the crds were updated like it did before the submodules?

Yes exactly, it works just like before which is awesome.

Regarding make update, I think just updating the apis submodule in main vendor dir would be ideal ,if it's possible, less likely to pull in unexpected things.

Reply:
> > > I tested the previous workflow with a make update, and was pleased to see the CRD yaml gets updated as before.
> > 
> > 
> > @dgoodwin i'm hoping this meant when you updated something in apis/ and ran `make update` the crds were updated like it did before the submodules?
> 
> Yes exactly, it works just like before which is awesome.
> 
> Regarding make update, I think just updating the apis submodule in main vendor dir would be ideal ,if it's possible, less likely to pull in unexpected things.

@dgoodwin overstepping the function of `go mod vendor` to make sure the vendor dir has the right contents seems like a bad idea to me. 

i could try adding a dependency on make update to check if there is diff in go submodule and run make vendor to update..